spec/lib/rewriters/logical_operator_assignment_spec.rb

```diff
@@ -0,0 +1,202 @@
+require 'lib/spec_helper'
+
+describe Opal::Rewriters::LogicalOperatorAssignment do
+  def s(type, *children)
+    ::Opal::AST::Node.new(type, children)
+  end
+
+  let(:rewriter) { Opal::Rewriters::LogicalOperatorAssignment.new }
+
+  def parse(source)
+    parser = Opal::Parser.default_parser
+    buffer = ::Parser::Source::Buffer.new('(eval)')
+    buffer.source = source
+    parser.parse(buffer)
+  end
+
+  def rewrite(ast)
+    rewriter.process(ast)
+  end
+
+  around(:each) do |e|
+    Opal::Rewriters::LogicalOperatorAssignment::AbstractHandler.reset_tmp_counter!
+    Opal::Rewriter.disable { e.run }
+  end
+  let(:cache_tmp_name) { :$or_equals_lhs_recvr_tmp_1 }
+  let(:cached) { s(:js_tmp, cache_tmp_name) }
+
+  shared_examples 'it rewrites' do |from:, to:|
```

Comment:
Sadly we still support ruby 2.0 which won't run this code, a bold move could be to skip it for the next release as it's doing rails I think… the first supporting this syntax should be 2.1